src/services/evento.js

```diff
@@ -4,20 +4,101 @@
 import Usuario from './usuario.js';
-export class Evento {
+export default class Evento {
     /**
      * 
-     * @param {String} Nombre // Nombre del Evento 
-     * @param {Arraylist(Usuario)} Participantes // Participantes que hay en un Pago
-     * @param {Arraylist(Pago)} Pagos // Lista De Pagos (Cada Pago contiene la cantidad y el usuario que la ha pagado)
+     * @param {String} nombre // Nombre del Evento 
+     * @param {Arraylist(Usuario)} participantes // Participantes que hay en un Evento
+     * @param {Arraylist(Pago)} pagos // Lista De Pagos (Cada Pago contiene la cantidad y el usuario que la ha pagado)
+     * @param {Dictionary(Usuario)} netos // Diccionario donde se apunta el estado actual de cada usuario
      *  
      */
-    Constructor(Nombre, Participantes) {
-        this.Nombre = Nombre;
-        this.Participantes = Participantes;
-        this.Pagos = [];
+    constructor(nombre, participantes) {
+        this.nombre = nombre;
+        this.participantes = participantes.slice();
+        this.pagos = [];
+        this.netos = new Map();
+        participantes.forEach(usuario => this.netos.set(usuario.getNombre(), 0))
     }
 
+    addPago(pago){
+        pago.getParticipantes().forEach(usuario => {
+            if(this.participantes.indexOf(usuario) == -1){
+                throw new Error(`El usuario "${usuario.getNombre()}" en el pago no está en el evento.`);
+            }
+        });
+        if((this.participantes.filter(usuario => usuario.getNombre() == pago.getUsuarioPaga().getNombre())).length == 0){
+            throw new Error(`El usuario "${pago.getUsuarioPaga().getNombre()}" que realiza el pago no está en el evento.`);
+        }
+
+        let debe = pago.getCantidad() / pago.getParticipantes().length;                                                                                          //Se divide el pago entre los participantes
+        pago.getParticipantes().forEach(usuario => this.netos.set(usuario.getNombre(), parseFloat((this.netos.get(usuario.getNombre()) + debe).toFixed(2))));    //Se suma a cada uno su parte
+        this.netos.set(pago.getUsuarioPaga().getNombre(), parseFloat((this.netos.get(pago.getUsuarioPaga().getNombre()) - pago.getCantidad()).toFixed(2)));      //Al que paga se le resta el importe
+        this.pagos.push(pago);
+    }
+
+    getNeto(usuario){
+        if(this.participantes.indexOf(usuario) == -1){
+            throw new Error(`El usuario "${usuario.getNombre()}" no está en el evento.`);
+        }
+        return this.netos[usuario]
+    }
+
+    getCuentas(){
+        let positivos = new Map();
+        let negativos = new Map();
+        let cuentas = [];
+        let cuentasFinal = [];
+
+        for(let [k,v] of this.netos.entries()){
+            if(v>0){
+                positivos.set(k,v)
+            }else if(v<0){
+                negativos.set(k,v)
+            }
+        }
+
+        let recorrer = (positivos, negativos, cuentas) => {
```

Comment:
¿Por qué es esto una variable local en vez de una función?

Reply:
Porque no puedo crear métodos privados en ES6 y no la utilizo desde ningún otro sitio.

Reply:
¿Y por qué usas ES6?
El problema con las funciones privadas es de documentación, sobre todo. Que no la uses desde ningún sitio no significa que no la vayas a usar; tendrías que haber planteado si la necesidad de esa función es específica o más genérica. Las buenas prácticas suelen pedir, además, funciones bastante compactas, que son más fáciles de comprender. El hecho de que no estés usando `this` apunta a que esto sería una función externa, no de clase. Podrías simplemetne no exportarla.

Reply:
Bueno, realmente no se si utilizo ES6, lo he dado por hecho por los imports y funciones flecha, pero intenté los métodos privados y no funcionaban, en cuando a donde la voy a utilizar tampoco la utilizaré desde fuera ya que necesito que dependa de una variable externa (cuentasFinal), por lo que pensándolo tampoco la podría implementar como un método privado ni externo, solamente dentro de esa función, anidándola. Pero no se si sería mejor implementarla como está ahora, o como una función anidada. (Entiendo que es lo mismo pero no lo sé)

Reply:
Dependerá de la versión de node que estás usando. Si estás usando la 18 (y no queda más remedio que lo hagas, por las bibliotecas de aserciones y test runners) se pueden usar perfectamente métodos privados.